api/src/main/java/com/datastrato/gravitino/rel/SupportsPartitions.java

```diff
@@ -79,21 +79,24 @@ default boolean partitionExists(String partitionName) {
   /**
    * Drop a partition with specified name.
    *
-   * @param partitionName The identifier of the partition.
-   * @return true if a partition was deleted, false if the partition did not exist.
+   * @param partitionName the name of the partition
+   * @param ifExists If true, will not throw NoSuchPartitionException if the partition not exists
+   * @return true if a partition was deleted.
    */
-  boolean dropPartition(String partitionName);
+  boolean dropPartition(String partitionName, boolean ifExists) throws NoSuchPartitionException;
```

Comment:
Missing the description of `NoSuchPartitionException`.

BTW, As described in https://github.com/datastrato/gravitino/issues/2434#issuecomment-2005680544, I am thinking that we should unify the behavior of dropping. 

Can we remove `ifExists` and `NoSuchPartitionException` here? That is, return true when deletion is successful, return false when the partition does not exist, or throw the corresponding exception otherwise.

Reply:
done